src/InMemoryRepository.php

```diff
@@ -255,6 +284,20 @@ private function addResource($path, Resource $resource)
         }
     }
 
+    private function moveResource(Resource $resource, $targetPath, &$moved)
+    {
+        $sourcePath = $resource->getPath();
+
+        foreach ($this->getChildIterator($resource) as $child) {
+            $this->moveResource($child, $targetPath.'/'.Path::getFilename($child->getPath()), $moved);
+        }
+
+        $this->resources[$targetPath] = $this->resources[$sourcePath];
```

Comment:
There is need for some validation here (if target path does not already exists) and also add rollback functionality.